yarn-project/end-to-end/src/e2e_zk_token_contract.test.ts

```diff
@@ -26,17 +26,16 @@ describe('e2e_zk_token_contract', () => {
   let accounts: AztecAddress[];
   let contract: Contract;
 
-  beforeAll(() => {
+  beforeAll(async () => {
     provider = createProvider(ETHEREUM_HOST, MNEMONIC, 1);
-  });
-
-  beforeEach(async () => {
     const ethRpc = new EthereumRpc(provider);
     logger('Deploying contracts...');
     rollupAddress = await deployRollupContract(provider, ethRpc);
     unverifiedDataEmitterAddress = await deployUnverifiedDataEmitterContract(provider, ethRpc);
     logger('Deployed contracts...');
+  });
```

Comment:
Why this change? Having a single shared contract instance will cause coupling across tests, which makes them a pain to debug when things go wrong.

Reply:
The issues in tree implementation were found from phils pr here: https://github.com/AztecProtocol/aztec3-packages/pull/233 where this change-set comes from.

It can make it a pain to debug but also exposed some issues

Reply:
Yeah, we can revert this back. It was just to illustrate the problem.